src/main/java/org/folio/services/piece/PieceService.java

```diff
@@ -84,18 +91,18 @@ public Future<Tx<PoLine>> updatePieces(Tx<PoLine> poLineTx, ReplaceInstanceRef r
 
   private Future<Tx<PoLine>> updateHoldingForPieces(Tx<PoLine> poLineTx, List<Piece> pieces, ReplaceInstanceRef replaceInstanceRef, DBClient client) {
     List<Piece> updatedPieces = new ArrayList<>();
-    List<Holding> holdings = replaceInstanceRef.getHoldings();
-    if (pieces == null) {
-      logger.info("Pieces wasn't updated");
+    if (CollectionUtils.isEmpty(pieces)) {
+      logger.info(PIECE_NOT_UPDATED, poLineTx.getEntity().getId());
       return Future.succeededFuture(poLineTx);
     }
-    holdings.forEach(holding -> updatedPieces.addAll(pieces.stream().filter(piece -> piece.getHoldingId().equals(holding.getFromHoldingId()))
-      .peek(piece -> {
-        if (holding.getToHoldingId() != null) {
+    replaceInstanceRef.getHoldings().forEach(holding -> updatedPieces.addAll(pieces.stream().filter(piece -> piece.getHoldingId().equals(holding.getFromHoldingId()))
+      .map(piece -> {
+        if (Objects.nonNull(holding.getToHoldingId())) {
           piece.setHoldingId(holding.getToHoldingId());
         } else {
           piece.setLocationId(holding.getToLocationId());
         }
+        return piece;
       })
       .collect(Collectors.toList())));
 
```

Comment:
Is it possible to add some unit test to cover issue described in the bug?

Reply:
Yeah sure, I'm working on it

Reply:
Done